src/repart/repart.c

```diff
@@ -7487,7 +7510,42 @@ static int context_fstab(Context *context) {
                 }
         }
 
-        r = flink_tmpfile(f, t, path, 0);
+        switch (arg_append_fstab) {
+        case APPEND_AUTO:
+                fputs(AUTOMATIC_FSTAB_HEADER_END "\n", f);
+
+                r = read_full_file(path, &c, &l);
+                if (r < 0) {
+                        if (r == -ENOENT) {
+                                log_warning("File fstab not found in %s", path);
+                                break;
+                        } else
+                                return log_error_errno(r, "Failed to open %s: %m", path);
+                }
+
+                if (startswith(c, "# Automatically generated by systemd-repart\n\n")) {
+                        log_warning("The fstab was generated by systemd-repart and will not be included in the new one. Ignoring");
+                        break;
+                } else {
+                        char *acs;
+                        const char *ace;
+                        acs = find_line(c, AUTOMATIC_FSTAB_HEADER_START);
+                        ace = find_line(c, AUTOMATIC_FSTAB_HEADER_END);
+                        if (acs != NULL)
+                                acs[-strlen(AUTOMATIC_FSTAB_HEADER_START) - 1] = '\0';
```

Comment:
uh, negative indexes?

Reply:
Right ... The thing here is that `find_line` return the position of the next line, so at least have `strlen() + 1` space before (the length plus '\n'). Should be expressed differently? Would more clear 'acs = strstr(AUTOMATIC_FSTAB_HEADER_START "\n");'?